app/code/Magento/CustomerGraphQl/Model/Customer/DeleteCustomer.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CustomerGraphQl\Model\Customer;
+
+use Magento\Customer\Api\CustomerRepositoryInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\GraphQl\Model\Query\ContextInterface;
+
+/**
+ * Get customer
+ */
+class DeleteCustomer
+{
+    /**
+     * @var CustomerRepositoryInterface
+     */
+    private $customerRepository;
+
+    /**
+     * @param CustomerRepositoryInterface $customerRepository
+     */
+    public function __construct(
+        CustomerRepositoryInterface $customerRepository
+    ) {
+        $this->customerRepository = $customerRepository;
+    }
+
+    /**
+     * Get customer
+     *
+     * @param ContextInterface $context
+     * @return void
+     * @throws GraphQlInputException
+     * @throws GraphQlNoSuchEntityException
+     */
+    public function execute(ContextInterface $context): void
```

Comment:
Why not simply passing the `$context->getUserId()` instead as customer ID?
```suggestion
    public function execute(int $customerId): void
```